modules/oguryBidAdapter.js

```diff
@@ -0,0 +1,151 @@
+
+  const isValidSizes = Boolean(adUnitSizes) && adUnitSizes.length > 0;
+  const isValidAdUnitId = !!bid.params.adUnitId;
+  const isValidAssetKey = !!bid.params.assetKey;
+
+  return (isValidSizes && isValidAdUnitId && isValidAssetKey);
+}
+
+function buildRequests(validBidRequests, bidderRequest) {
+  const openRtbBidRequestBanner = {
+    id: bidderRequest.auctionId,
+    tmax: DEFAULT_TIMEOUT,
+    at: 2,
+    regs: {
+      ext: {
+        gdpr: 1
+      },
+    },
+    site: {
+      domain: location.hostname
+    },
+    user: {
+      ext: {
+        consent: ''
+      }
+    },
+    imp: []
+  };
+
+  if (bidderRequest.hasOwnProperty('gdprConsent') &&
+    bidderRequest.gdprConsent.hasOwnProperty('gdprApplies')) {
+    openRtbBidRequestBanner.regs.ext.gdpr = bidderRequest.gdprConsent.gdprApplies ? 1 : 0
+  }
+
+  if (bidderRequest.hasOwnProperty('gdprConsent') &&
+    bidderRequest.gdprConsent.hasOwnProperty('consentString') &&
+    bidderRequest.gdprConsent.consentString.length > 0) {
+    openRtbBidRequestBanner.user.ext.consent = bidderRequest.gdprConsent.consentString
+  }
+
+  validBidRequests.forEach((bidRequest) => {
+    const sizes = getAdUnitSizes(bidRequest)
+      .map(size => ({ w: size[0], h: size[1] }));
+
+    if (bidRequest.hasOwnProperty('mediaTypes') &&
+      bidRequest.mediaTypes.hasOwnProperty('banner')) {
+      openRtbBidRequestBanner.site.id = bidRequest.params.assetKey;
+
+      openRtbBidRequestBanner.imp.push({
+        id: bidRequest.bidId,
+        tagid: bidRequest.params.adUnitId,
+        bidfloor: getFloor(bidRequest),
+        banner: {
+          format: sizes
+        }
+      });
+    }
+  });
+
+  return {
+    method: 'POST',
+    url: BID_HOST,
+    data: openRtbBidRequestBanner
+  };
+}
+
+function interpretResponse(openRtbBidResponse) {
+  if (!openRtbBidResponse ||
+    !openRtbBidResponse.body ||
+    typeof openRtbBidResponse.body != 'object' ||
+    Object.keys(openRtbBidResponse.body).length === 0) {
+    logWarn('no response or body is malformed');
+    return [];
+  }
+
+  const bidResponses = [];
+
+  openRtbBidResponse.body.seatbid.forEach((seatbid) => {
+    seatbid.bid.forEach((bid) => {
+      let bidResponse = {
+        requestId: bid.impid,
+        cpm: bid.price,
+        currency: 'USD',
+        width: bid.w,
+        height: bid.h,
+        creativeId: bid.id,
+        netRevenue: true,
+        ttl: 60,
+      };
+
+      bidResponse.ad = bid.adm;
+
+      bidResponses.push(bidResponse);
+    });
+  });
+
+  if (bidResponses.length === 0) logWarn('bid responses is empty');
+
+  return bidResponses;
+}
+
+function onTimeout(timeoutData) {
+  logWarn(timeoutData);
+}
+
+function onBidWon(bid) {
+}
+
+function onSetTargeting(bid) {
+  logWarn(bid);
+}
+
+function getFloor(bid) {
+  if (isFn(bid.getFloor)) {
+    let floorResult = bid.getFloor({
+      currency: 'USD',
+      mediaType: 'banner',
+      size: '*'
+    });
+
+    return floorResult.floor;
```

Comment:
it is possible that if floors is set in non USD, and publisher does not have currency module included / enabled, that this call will return non USD floor.

It is a good idea to verify that the currency returned is in USD in this case, just to be safe!

Maybe something like this?
```js
return floorResult.currency === 'USD' ? floorResult.floor : 0;
```

Reply:
I agree. I added the check you suggested and a new unit test to cover the case of unsupported currency 
it("should handle bidFloor when currency is not USD")